add dts for different configurations

Radxa-Alvin opened the pull request.

Radxa-Alvin committed:
arm: dts: rock 3c: add dts for different configurations

The purpose of adding dts with different configurations is to
allow U-BOOT to start different configurations based on HW_ID.

Signed-off-by: Alvin Xie <alvin@radxa.com>

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/Makefile:
```diff
@@ -126,10 +126,14 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rk817-tablet-k108.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rk817-tablet-rkg11.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rk817-tablet-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-evb1-ddr4-v10.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rock-3c-aic8800ds2.dtbo
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rock-3c-ap6256.dtbo
```
是dtb

Radxa-Alvin replied:
目前没修改原来的DTS  新增了 去除差异配置的 DTSI，原来的DTS 还是保持不动吧  这样老的板子不会有影响

RadxaYuntian replied:
主要是我们时不时会有一些设备树的修改，现在你这样就需要改两个地方

Radxa-Alvin replied:
把目前的rock3c.dts改为 rock3c.dtsi 再新增一个空白的rock3c.dts引用rock3c.dtsi吗

RadxaYuntian replied:
3c.dtsi是共通部分，里面是没有wifi定义的。ap wifi定义放到rock-3c.dts

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/Makefile:
```diff
@@ -126,10 +126,14 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rk817-tablet-k108.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rock-3c.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-radxa-cm3-sodimm-io.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-radxa-cm3s-io.dtb
-dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-radxa-zero3.dtb
```
原来的zero3.dtb必须保留，不然现有用户升级后无法启动

RadxaYuntian reviewed (changes requested):
另外现有的zero3.dts和3c.dts需要改成使用对应的dtsi以及配置好当前所支持的wifi/以太网，避免代码重复

Radxa-Alvin committed:
arm: dts: radxa zero3: add dts for different configurations

The purpose of adding dts with different configurations is to
allow U-BOOT to start different configurations based on HW_ID.

Signed-off-by: Alvin Xie <alvin@radxa.com>

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts:
```diff
@@ -0,0 +1,49 @@
+
+/ {
+	model = "Radxa ZERO 3";
+	compatible = "radxa,zero-3e","radxa,zero3", "rockchip,rk3566";
```
```suggestion
	compatible = "radxa,zero-3e", "radxa,zero3", "rockchip,rk3566";
```

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-aic8800ds2.dts:
```diff
@@ -0,0 +1,150 @@
+&mdio1 {
+	status = "disabled";
+};
+
```
这些应该在dtsi里面就禁用了，然后只在3e里面打开

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-ap6212.dts:
```diff
@@ -0,0 +1,165 @@
+&mdio1 {
+	status = "disabled";
+};
+
```
3e

RadxaYuntian reviewed (changes requested):
另外需要把zero3.dts也仿照3c.dts改成使用dtsi

RadxaYuntian commented:
以太网的定义应该只放在3e.dts而不是zero3.dtsi
zero3.dts也没转换成使用zero3.dtsi

Radxa-Alvin commented:
前面不是说把配置写 dtsi 么然后在 3e enable么，原来的DTS 同时使能了 wifi 和以太网，现在的dtsi 把以太网相关的节点先关掉了，有点差异，先这样用着吧，后面板子硬件有修改再做改动吧

RadxaYuntian commented:
说的是dtsi不写以太网也不写wifi
